docs/sdk-json-proposal/SDK_API_DEFINITION.md

```diff
@@ -0,0 +1,251 @@
+# Mesh Data Plane SDK Definition
+
+## mesh_create_client()
+```c
+int mesh_create_client(MeshClient **client,
+                       const char *config_json)
+```
+Creates a new mesh client with the provided JSON configuration.
+
+#### Parameters
+* `[OUT]` `client` – Address of a pointer to a mesh client structure.
+* `[IN]` `config_json` – Pointer to a mesh client configuration string in JSON format.
+
+#### Client Configuration Example
+```json
+{
+  "apiVersion": "v1",
+  "apiConnectionString": "Server=192.168.95.1; Port=8001",
+  "apiDefaultTimeoutMicroseconds": 100000,
+  "maxMediaConnections": 32,
+}
+```
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_delete_client()
+```c
+int mesh_delete_client(MeshClient **client)
+```
+Deletes the mesh client and its resources.
+
+#### Parameters
+* `[IN/OUT]` `client` – Address of a pointer to a mesh client structure.
+
+#### Returns
+0 if successful. Otherwise, returns an error.
+
+
+## mesh_create_tx_connection()
+```c
+int mesh_create_tx_connection(MeshClient *client,
+                              MeshConnection **conn,
+                              const char *config_json)
+```
+Creates a transmitter connection with the provided JSON configuration.
+
+#### Parameters
+* `[IN]` `client` – Pointer to a parent mesh client.
+* `[OUT]` `conn` – Address of a pointer to the connection structure.
+* `[IN]` `config_json` – Pointer to a connection configuration structure.
+
+#### Transmitter/Receiver Connection Configuration Example
+```json
+{
+  "bufferQueueCapacity": 16,
+  "maxPayloadSize": 2097152,
+  "maxMetadataSize": 8192,
+  "connection": {
+    "multipoint-group": {
```

Comment:
I think multipoint-group should be the only option possible.

Reply:
Yes, from App perspective the only way of communication is with multipoint group. But we need to have ability to pass additional parameters for transports like ST2110 or RDMA which are come from NMOS controller and to override default values. They are to be passed to MCM Agent and then MCM Agent is to process them and send configuration to proper instance of ST2110 or RDMA bridge.

Reply:
So I would leave it as it is. To have multipoint-group as mandatory connection type and mandatory corresponding section. And in parallel to allow optionally to add configuration for st2110 and RDMA and potentially other in the future.

Reply:
> MCM Agent is to process them and send configuration to proper instance of ST2110 or RDMA bridge.
@pwysock1-intel This is the business logic that you need to define in detail: how the Agent decides which instance is `proper`.